src/Microsoft.Identity.Client/TokenCacheNotificationArgs.cs

```diff
@@ -5,25 +5,37 @@ namespace Microsoft.Identity.Client
 {
     /// <summary>
     /// Contains parameters used by the MSAL call accessing the cache.
-    /// See also <see cref="T:ITokenCache"/> which contains methods
+    /// See also <see cref="T:ITokenCacheSerializer"/> which contains methods
     /// to customize the cache serialization
     /// </summary>
     public sealed partial class TokenCacheNotificationArgs
     {
+        internal TokenCacheNotificationArgs(
+            ITokenCacheSerializer tokenCacheSerializer,
+            string clientId,
+            IAccount account,
+            bool hasStateChanged)
+        {
+            TokenCache = tokenCacheSerializer;
+            ClientId = clientId;
+            Account = account;
+            HasStateChanged = hasStateChanged;
+        }
+
         /// <summary>
-        /// Gets the <see cref="ITokenCache"/> involved in the transaction
+        /// Gets the <see cref="ITokenCacheSerializer"/> involved in the transaction
         /// </summary>
-        public ITokenCache TokenCache { get; internal set; }
+        public ITokenCacheSerializer TokenCache { get; }
```

Comment:
I guess we used the same name to avoid people to have to change a lot of code?

Reply:
Yes, that's correct.  For people who already are serializing in their callbacks, this fix will "just work" which is ideally most of the cases.

Reply:
Thanks for confirming. Awesome. The breaking change is really making sure that people do the right thing.

Reply:
nice.

---
In reply to: [287077505](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/1170#discussion_r287077505) [](ancestors = 287077505)